test/combine_latest_test.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:pedantic/pedantic.dart';
+import 'package:test/test.dart';
+
+import 'package:stream_transform/stream_transform.dart';
+
+void main() {
+  group('combineLatest', () {
+    test('flows through combine callback', () async {
+      var source = StreamController<int>();
+      var combineWith = StreamController<int>();
+      int sum(int a, int b) => a + b;
+
+      var results = <int>[];
+      unawaited(source.stream
+          .transform(combineLatest(combineWith.stream, sum))
+          .forEach(results.add));
+
+      source.add(1);
+      await Future(() {});
```

Comment:
Why not pumpEventQueue? https://docs.flutter.io/flutter/package-test_api_test_api/pumpEventQueue.html

Reply:
`pumpEventQueue` is good for the cases where the work that's happening may create additional trips around the event loop while it's running. In this case we wouldn't want that to happen (though, admittedly, we aren't really being precise about guaranteeing it)

I'll leave this as is for now, if we decide to switch over to `pumpEventQueue` we'll probably do all the tests at once.